docs/src/main/asciidoc/config-reference.adoc

```diff
@@ -674,19 +711,51 @@ extensions. Therefore, the `quarkus.` prefix should **never** be used for applic
 
 === Build Time configuration
 
-Some Quarkus configurations only take effect during build time, meaning it is not possible to change them at runtime. These
-configurations are still available at runtime but as read-only and have no effect in Quarkus behaviour. A change to any
-of these configurations requires a rebuild of the application itself to reflect changes of such properties.
+Some Quarkus configurations only take effect during build time (compile), meaning it is not possible to change it at
```

Comment:
Maybe we should make it clear that it's not about application configuration but core and extensions? But I don't have the structure of the document in mind so maybe it's already clear.

Reply:
This is a subsection of:

https://github.com/quarkusio/quarkus/blob/084cc958b4da0dfa6adc6a392521ea33c2d716a3/docs/src/main/asciidoc/config-reference.adoc#L700-L704

So I think it should be covered.